summingbird-core-test/src/test/scala/com/twitter/summingbird/memory/MemoryLaws.scala

```diff
@@ -202,6 +202,18 @@ class MemoryLaws extends WordSpec {
     "lookupCollect w/ Int, Int" in { assert(lookupCollectChecker[Int, Int] == true) }
 
     "counters w/ Int, Int, Int" in { assert(counterChecker[Int, Int, Int] == true) }
+
+    "needless also shouldn't cause a problem" in {
+      val source = Memory.toSource(0 to 100)
+      val store1 = MutableMap.empty[Int, Int]
+      val store2 = MutableMap.empty[Int, Int]
+      val comp = source.map(v => (v % 3, v)).sumByKey(store1)
+      val prod = comp.also(comp.mapValues(_._2).write(new BufferFunc).sumByKey(store2))
```

Comment:
I tried a bunch of stuff (which is why it is not trivial example up there). I started with `comp.also(comp)` and expanded out. I just ran it with an extra `.also(comp)` and that passed for me as well.

I think there is a chance he had a service backed by the same map somewhere in the job, which was the real issue (but I'm just guessing).